packages/jupyter-ai-magics/jupyter_ai_magics/magics.py

```diff
@@ -578,12 +589,29 @@ def run_ai_cell(self, args: CellArgs, prompt: str):
         prompt = prompt.format_map(FormatDict(ip.user_ns))
 
         if provider.is_chat_provider:
-            result = provider.generate([[HumanMessage(content=prompt)]])
+            result = provider.generate(
+                [[*self.transcript, HumanMessage(content=prompt)]]
+            )
         else:
             # generate output from model via provider
-            result = provider.generate([prompt])
+            if self.transcript:
+                transcript = [
+                    (
+                        f"<HUMAN>{message.content}</HUMAN>"
+                        if message.type == "human"
+                        else message.content
+                    )
+                    for message in self.transcript + [HumanMessage(content=prompt)]
```

Comment:
I'm not familiar with this pseudo-XML syntax, and it may cause confuse LLMs to also return their responses in pseudo-XML, e.g. `<AI>The square root of pi is...</AI>`.

Can a more conventional string formatting be used instead here? LangChain simply prepends `Human: ` to human messages and `AI: ` to AI messages.